### PR TITLE
Qualify Olive INT2 checkpoint export

### PR DESCRIPTION
**Pull request by titaiwangms**
## Summary

- qualify synthetic Olive-format uniform INT2 checkpoints through the public local `build()` path
- qualify mixed default INT2 with INT4 `down_proj` and INT8 `gate_proj` overrides
- save and reload the Mobius package, then verify exact `MatMulNBits` attributes, input wiring, packed bytes, scales, and initializer binding
- execute representative emitted INT2/INT4/INT8 projections explicitly on `CPUExecutionProvider` against an independent Olive-compatible decode

## Scope

This validates the Olive checkpoint format/import/export contract and CPU operator semantics. It does **not** claim CUDA INT2 kernel readiness or performance, and it does not cover QMoE/fused MoE. No Mobius production changes were required after #733.

## Validation

- `python -m pytest src/mobius/_component_quantization_loading_test.py -q --tb=short` (61 passed)
- `python -m ruff check src/mobius/_component_quantization_loading_test.py`
- `python -m ruff format --check src/mobius/_component_quantization_loading_test.py`
- `git diff --check`

Related: microsoft/onnxruntime#32657, microsoft/Olive#2669

### CODE REVIEW

File: src/mobius/_component_quantization_loading_test.py [hunk 1]
```diff
@@ -449,6 +538,219 @@ def test_public_build_loads_global_olive_rules_without_component_plan(
...
+        olive_sidecars[name] = (qweight, scales)
+    assert not any("qzero" in name for name in checkpoint)
+    save_file(checkpoint, str(checkpoint_dir / "model.safetensors"))
+
+    # This is the public local-checkpoint path: config discovery, safetensors
+    # loading, source-name adaptation, layout normalization, binding, and export.
+    package = build(str(checkpoint_dir), dtype="f32")
+    export_dir = tmp_path / "exported"
+    package.save(str(export_dir), progress_bar=False, max_workers=1)
+    graph = ir.load(export_dir / "model.onnx").graph
```
[the remark this answers is not in the document]
Comment (titaiwangms) [reply src/mobius/_component_quantization_loading_test.py]: Fixed in 4b927b35. The test now reloads the saved export through `ModelPackage.load(export_dir)["model"]`, so all graph, binding, and CPU projection assertions run against the package-level round trip. The full co-located test file still passes (61 tests).

**Comment by titaiwangms** [comment]
Cross-repository real-model qualification completed locally with an Olive-produced RTN INT2 checkpoint from `hf-internal-testing/tiny-random-LlamaForCausalLM`, pinned at `9fb191250dd56d0ba7ec9785a025ed29c03d5998` (~1.03M parameters).

- Mobius consumed the Olive output directory unchanged.
- `ModelPackage.save` / `ModelPackage.load` produced 14 `com.microsoft::MatMulNBits(bits=2, block_size=16)` projections across both layers.
- Every exported packed-weight initializer and scale initializer matched the source Olive safetensors exactly.
- A representative CPU `MatMulNBits` projection had max absolute error `0.0` against an independent decode.
- The complete exported ONNX model loaded and executed on `CPUExecutionProvider`; logits for a 3-token prompt matched Olive quantized PyTorch with max absolute error `8.940696716308594e-08`.

No model artifacts were committed. This remains CPU/format qualification, not CUDA INT2 performance validation.

**Comment by titaiwangms** [comment]
Real-model SMP → RTN mixed-precision qualification also completed with the same pinned tiny Llama revision.

- Mobius consumed the Olive checkpoint unchanged and exported 14 `MatMulNBits` nodes: 11 INT2, 1 INT4, and 2 INT8.
- Every packed-weight and scale initializer matched the Olive safetensors exactly.
- Representative INT2, INT4, and INT8 CPU projections each had max absolute error `0.0` against independent decoding.
- Full-model CPU logits matched the Olive quantized PyTorch model with max absolute error `8.940696716308594e-08`.

The actual SMP heuristic selected only layer 0 `down_proj` for INT4 on this 2-layer model; the two INT8 `gate_proj` entries came from explicit RTN overrides. No artifacts were committed.

**Review by gramalingam — APPROVED** [review]
